test/src/micro/attributes_test.dart

```diff
@@ -11,37 +11,56 @@ import 'package:test/test.dart';
 import 'package:polymer/polymer.dart';
 import 'package:smoke/mirrors.dart' as smoke;
 
-AttributesTest element;
+const _attributesName = 'attributes-test';
+const _serializeAttributesName = 'serialize-attributes-test';
 
-main() async {
-  smoke.useMirrors();
-  await initPolymer();
+_tests(String elementName) {
+  group(elementName, () {
+    var element;
 
-  setUp(() {
-    element = querySelector('attributes-test');
-  });
+    setUp(() {
+      element = querySelector(elementName);
+    });
 
-  test('attributes can be marshalled into properties', () {
-    expect(element.myString, 'string');
-    expect(element.myNum, 2);
-    expect(element.myBool, true);
-    expect(element.myMap, {'hello': 'world'});
-    expect(element.myList, ['hello', 'world']);
-    expect(element.myDateTime, new DateTime(1987, 07, 18));
-  });
+    test('attributes can be marshalled into properties', () {
+      expect(element.myString, 'string');
+      expect(element.myNum, 2);
+      expect(element.myBool, true);
+      expect(element.myMap, {'hello': 'world'});
+      expect(element.myList, ['hello', 'world']);
+      expect(element.myDateTime, new DateTime(1987, 07, 18));
+    });
+
+    test('hostAttributes are applied to the host', () {
+      expect(element.attributes['host-string'], 'string');
+      expect(element.attributes['host-num'], '2');
+      expect(element.attributes['host-bool'], '');
+      expect(element.attributes['host-map'], '{"hello":"world"}');
+      expect(element.attributes['host-list'], '["hello","world"]');
+  //    expect(DateTime.parse(element.attributes['host-date-time']),
+  //        new DateTime(1987, 07, 18));
+    });
 
-  test('hostAttributes are applied to the host', () {
-    expect(element.attributes['host-string'], 'string');
-    expect(element.attributes['host-num'], '2');
-    expect(element.attributes['host-bool'], '');
-    expect(element.attributes['host-map'], '{"hello":"world"}');
-    expect(element.attributes['host-list'], '["hello","world"]');
-//    expect(DateTime.parse(element.attributes['host-date-time']),
-//        new DateTime(1987, 07, 18));
+    if (elementName == _serializeAttributesName) {
```

Comment:
I just had it in there to keep it in the same group. I'm fine with moving it around if you don't like. Seemed like less typing :wink: